monkey/infection_monkey/config.py

```diff
@@ -18,57 +16,50 @@
 EXTERNAL_CONFIG_FILE = os.path.join(os.path.abspath(os.path.dirname(sys.argv[0])), 'monkey.bin')
 
 
-def _cast_by_example(value, example):
-    """
-    a method that casts a value to the type of the parameter given as example
-    """
-    example_type = type(example)
-    if example_type is str:
-        return os.path.expandvars(value).encode("utf8")
-    elif example_type is tuple and len(example) != 0:
-        if value is None or value == tuple([None]):
-            return tuple()
-        return tuple([_cast_by_example(x, example[0]) for x in value])
-    elif example_type is list and len(example) != 0:
-        if value is None or value == [None]:
-            return []
-        return [_cast_by_example(x, example[0]) for x in value]
-    elif example_type is type(value):
-        return value
-    elif example_type is bool:
-        return value.lower() == 'true'
-    elif example_type is int:
-        return int(value)
-    elif example_type is float:
-        return float(value)
-    elif example_type in (type, ABCMeta):
-        return globals()[value]
-    else:
-        return None
+class Configuration(object):
 
+    def from_kv(self, formatted_data):
+        # now we won't work at <2.7 for sure
+        network_import = importlib.import_module('infection_monkey.network')
+        exploit_import = importlib.import_module('infection_monkey.exploit')
 
-class Configuration(object):
-    def from_dict(self, data):
-        """
-        Get a dict of config variables, set known variables as attributes on self.
-        Return dict of unknown variables encountered.
-        """
-        unknown_variables = {}
-        for key, value in data.items():
+        unknown_items = []
+        for key, value in formatted_data.items():
             if key.startswith('_'):
                 continue
             if key in ["name", "id", "current_server"]:
                 continue
             if self._depth_from_commandline and key == "depth":
                 continue
-            try:
-                default_value = getattr(Configuration, key)
-            except AttributeError:
-                unknown_variables[key] = value
-                continue
-
-            setattr(self, key, _cast_by_example(value, default_value))
-        return unknown_variables
+            if 'class' in key:
```

Comment:
I don't see the point of this main if. In the future we might have another config value which requires this intervention and not have the word 'class' in it